backend/src/achievements-log/dto/achievements-log.dto.ts

```diff
@@ -0,0 +1,24 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty } from 'class-validator';
+
+export class AchievementsLogDto {
```

Comment:
Genuinely naive question here, but shouldn't the logDto contains the achievement dto or some of its information ? Do we expect the client to send two requests everytime ? After all it is the achievement entity that contains the data we might want to display most (title, image...)

Reply:
This is a good question, and I think you have the right idea here

Normally I push for sending the `id` instead of the object, to avoid repeatedly sending the same object to a client that already has a copy stored. But I don't think the client will be storing all of the achievements, so that shouldn't be an issue in this case